framework/src/problems/FEProblemBase.C

```diff
@@ -3517,6 +3517,7 @@ FEProblemBase::addUserObject(const std::string & user_object_name,
     auto suo = std::dynamic_pointer_cast<SideUserObject>(user_object);
     auto isuo = std::dynamic_pointer_cast<InternalSideUserObject>(user_object);
     auto iuo = std::dynamic_pointer_cast<InterfaceUserObject>(user_object);
+    auto iuob = std::dynamic_pointer_cast<InterfaceUserObjectBase>(user_object);
```

Comment:
```suggestion
    auto iuob = std::dynamic_pointer_cast<InterfaceUserObjectBase>(user_object);
```

Reply:
if you can cast to `InterfaceUserObject` it's guaranteed that you can cast to `InterfaceUserObjectBase` so that's redundant